MetadataExtractor/Formats/Apple/BplistReader.cs

```diff
@@ -36,17 +36,17 @@ public static bool IsValid(byte[] bplist)
         return true;
     }
 
-    public static PropertyListResults Parse(byte[] bplist)
+    public static PropertyListResults Parse(ReadOnlySpan<byte> bplist)
     {
         if (!IsValid(bplist))
         {
             throw new ArgumentException("Input is not a bplist.", nameof(bplist));
         }
 
-        Trailer trailer = ReadTrailer();
+        Trailer trailer = ReadTrailer(bplist);
 
         int offset = checked((int)(trailer.OffsetTableOffset + trailer.TopObject));
-        var reader = new BufferReader(bplist.AsSpan(offset), isBigEndian: true);
+        var reader = new BufferReader(bplist.Slice(offset), isBigEndian: true);
 
         int[] offsets = new int[(int)trailer.NumObjects];
```

Comment:
The loop below here (can't comment there) can also be improved with something like:

```c#
        if (trailer.OffsetIntSize == 1)
        {
            reader.GetBytes(offsets);
        }
        else if (trailer.OffsetIntSize == 2)
        {
            for (int i = 0; i < offsets.Length; i++)
            {
                offsets[i] = reader.GetUInt16();
            }
        }
```

Reply:
We're relying on an implicit upcast from byte to int here -- so we can't read directly. However, let me refactor to avoid the extra calls / bounds checks.